src/main/java/week3/Dictionary.java

```diff
@@ -0,0 +1,35 @@
+package week3;
+
+
+import java.util.Arrays;
+
+public class Dictionary {
+
+  private String[] words;
+
+  Dictionary(String[] words) {
+    this.words = words;
+  }
+
+  /*
+  * Time Complexity : O(N) where N is the number of words in your dictionary
+  */
+  public boolean isWord(String word) {
+    if (words == null || word == null) {
+      return false;
+    }
+
+    return Arrays.stream(words).filter(w -> w.equals(word)).findAny().isPresent();
+  }
+
+  /*
+   * Time Complexity : O(N) where N is the number of words in your dictionary
+   */
+  public boolean isPrefix(String prefix) {
+    if (words == null || prefix == null) {
+      return false;
+    }
+
+    return Arrays.stream(words).filter(w -> w.indexOf(prefix) == 0).findAny().isPresent();
+  }
+}
```

Comment:
For a quick solution, I wouldn't even bother with a trie. Create one set for words, and one set for prefixes, and just populate the prefix set with all the prefixes of the dictionary words. Trie works fine (and allows extra optimisations), but is harder work.